smooch.go

```diff
@@ -124,6 +142,25 @@ func (sc *smoochClient) Handler() http.Handler {
 	return sc.mux
 }
 
+// IsJWTExpired will check whether Smooch JWT is expired or not.
+func (sc *smoochClient) IsJWTExpired() (bool, error) {
+	return isJWTExpired(sc.jwtToken, sc.secret)
+}
+
+// RenewToken will generate new Smooch JWT token.
+func (sc *smoochClient) RenewToken() (string, error) {
+	sc.mtx.Lock()
+	defer sc.mtx.Unlock()
+
+	jwtToken, err := GenerateJWT("app", sc.keyID, sc.secret)
```

Comment:
Before generating new token - a check is needed, to verify that it wasn't generated previously.